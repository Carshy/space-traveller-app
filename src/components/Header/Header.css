.navigation-section {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 44px;
  width: 100%;
  height: 8rem;
  box-shadow: rgba(25, 25, 25, 0.04) 0 0 1px 0, rgba(0, 0, 0, 0.1) 0 3px 4px 0;
  margin-bottom: 3%;
}

.nav-group {
  display: flex;
  list-style-type: none;
  gap: 44px;
  margin-right: 4%;
}

.nav-item {
  font-size: 1.5rem;
  color: #45a3ff;
  text-decoration: none;
}

.nav-link {
  display: flex;
  flex-direction: row;
  align-items: center;
  color: #000;
  font-weight: bold;
  text-decoration: none;
  margin-left: 4%;
}

.rocket-logo {
  width: 6.2rem;
  height: auto;
}

.text-logo {
  font-size: 2rem;
}

.nav-item:focus {
  color: #000;
  text-decoration: underline;
}
