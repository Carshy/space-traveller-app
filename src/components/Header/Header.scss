.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 44px;
  width: 100%;
  /* box-shadow: rgba(25, 25, 25, 0.04) 0 0 1px 0, rgba(0, 0, 0, 0.1) 0 3px 4px 0; */
  position: fixed;
  z-index: 11;
  background: transparent;
}

.navbar.fixed {
  // box-shadow: rgba(25, 25, 25, 0.04) 0 0 1px 0, rgba(0, 0, 0, 0.1) 0 3px 4px 0;
  backdrop-filter: blur(189px);
  background-color: #fff;
  transition: all 0.3s ease-in-out;

  .nav-link {
    .text-logo {
      color: #000;
    }
  }

  ul {
    li {
      .nav-item {
        color: #000;
      }
    }
  }
}

.nav-group {
  display: flex;
  list-style-type: none;
  gap: 44px;
  margin-right: 4%;

  @media screen and(max-width: 850px) {
    display: none;
  }
}

.nav-item {
  font-size: 1.5rem;
  color: #fff;
  // color: #45a3ff;
  text-decoration: none;
}

.nav-link {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  color: #fff;
  font-weight: bold;
  padding: 1rem 0 1rem 2rem;

  img {
    width: 80px;
    height: auto;
    margin-right: 12px;

    @media screen and(min-width: 2000px) {
      width: 180px;
      height: 40px;
    }

    @media screen and(max-width: 850px) {
      width: 60px;
      height: auto;
    }
  }
}

.app__navbar-menu {
  width: 36px;
  height: 36px;
  position: relative;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  background: transparent;
  // padding-right: 5rem;

  svg {
    width: 75%;
    height: 75%;
    color: #fff;
  }

  div {
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    z-index: 5;
    padding: 1rem;
    width: 80%;
    height: 100vh;
    display: flex;
    justify-content: flex-end;
    align-items: flex-end;
    flex-direction: column;
    background: url("../../images/bg4.jpeg");
    background-size: cover;
    background-repeat: repeat;
    background-color: #fff;
    box-shadow: 0 0 20px rgba(168, 168, 168, 0.8);

    svg {
      width: 34px;
      height: 34px;
      color: #fff;
      margin: 0.5rem 1rem;
    }

    ul {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
      display: flex;
      justify-content: flex-start;
      align-items: flex-start;
      flex-direction: column;

      li {
        list-style-type: none;
        margin: 1rem;

        .nav-item {
          text-decoration: none;
          color: #fff;
          font-size: 1rem;
          text-transform: uppercase;
          font-weight: bold;
          transition: all 0.3s ease-in-out;
          font-size: 1.2rem;

          &:hover {
            color: #8dc6ff;
          }
        }
      }
    }

    @media screen and(min-width: 8500px) {
      display: none;
    }
  }
  @media screen and(min-width: 850px) {
    display: none;
  }
}

.text-logo {
  font-size: 2rem;

  @media screen and(max-width: 850px) {
    font-size: 1.2rem;
  }
}

.active {
  text-decoration: underline;
}
