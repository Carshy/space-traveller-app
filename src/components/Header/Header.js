import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.scss';
import planetLogo from './planetLogo.png';

function Navigation() {
  return (
    <div>
      <nav className="navigation-section">
        <a className="nav-link" href="/">
          <img className="rocket-logo" src={planetLogo} alt="planet logo" />
          <span className="text-logo">Space Hub Shuttle</span>
        </a>
        <ul className="nav-group">
          <li>
            <NavLink to="/" className="nav-item">
              Home
            </NavLink>
          </li>
          <li>
            <NavLink to="/rockets" className="nav-item">
              Rockets
            </NavLink>
          </li>
          <li>
            <NavLink to="/missions" className="nav-item">
              Missions
            </NavLink>
          </li>
          <li>
            <NavLink to="/profiles" className="nav-item">
              My Profile
            </NavLink>
          </li>
        </ul>
      </nav>
    </div>
  );
}

export default Navigation;
