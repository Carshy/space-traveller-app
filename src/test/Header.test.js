import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Renderer from 'react-test-renderer';
import Header from '../components/Header/Header';

function MockHeader() {
  return (
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  );
}

it('Renders correctly', () => {
  const tree = Renderer
    .create(<MockHeader />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
