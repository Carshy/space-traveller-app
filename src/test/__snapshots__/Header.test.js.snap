// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <nav
    className="navigation-section"
  >
    <a
      className="nav-link"
      href="/"
    >
      <img
        alt="planet logo"
        className="rocket-logo"
        src="planetLogo.png"
      />
      <span
        className="text-logo"
      >
        Space Travelers' Hub
      </span>
    </a>
    <ul
      className="nav-group"
    >
      <li>
        <a
          aria-current="page"
          className="nav-item active active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className="nav-item"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className="nav-item"
          href="/profiles"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
